pkgs/development/python-modules/pypass/default.nix

```diff
@@ -0,0 +1,89 @@
+{ buildPythonPackage
+, click
+, colorama
+, enum34
+, fetchFromGitHub
+, fetchgit
+, fetchPypi
+, git
+, gnugrep
+, gnupg
+, nose
+, pbr
+, pexpect
+, pythonAtLeast
+, pythonOlder
+, stdenv
+, tree
+, xclip
+}:
+
+# NOTE: pypass can also be used as an application, but probably the most
+# important usecase is as a library. So, let's use buildPythonPackage and
+# support any Python version instead of defining it as an application with
+# buildPythonApplication.
+buildPythonPackage rec {
+  pname = "pypass";
+  version = "0.2.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1nm4mj7pd7gz4ghic6b3wrnd1b59hd1f0axavdabfl79wy511l7r";
+  };
+
+  patches = [ ./mark-executables.patch ];
+
+  # Paths to executables that pypass uses
+  git_exec = "${git}/bin/git";
+  grep_exec = "${gnugrep}/bin/grep";
+  gpg_exec = "${gnupg}/bin/gpg2";
+  tree_exec = "${tree}/bin/tree";
+  xclip_exec = "${xclip}/bin/xclip";
+
+  # Remove enum34 requirement if Python >= 3.4. Fix the paths to some
+  # executables that the Python code runs as subprocesses.
+  postPatch = ''
+    substituteAllInPlace Makefile
+    substituteAllInPlace pypass/command.py
+    substituteAllInPlace pypass/passwordstore.py
+    substituteAllInPlace pypass/tests/test_command.py
+    substituteAllInPlace pypass/tests/test_passwordstore.py
+  '' + stdenv.lib.optionalString (pythonAtLeast "3.4") ''
+    substituteInPlace requirements.txt --replace "enum34" ""
+  '';
+
+  buildInputs = [ pbr ];
+
+  propagatedBuildInputs = [
+    click
+    colorama
+    pexpect
+  ] ++ stdenv.lib.optionals (pythonOlder "3.4") enum34;
```

Comment:
It probably is not worth it supporting such old versions.

Reply:
nixpkgs still contains Python 2 support..

Reply:
I didn't remove this but I fixed `optionals` to `optional`. I don't see any harm in supporting Python 2 as long as it's kept available in nixpkgs.